crates/scmi/src/devices/common.rs

```diff
@@ -0,0 +1,531 @@
+    fn number_of_axes(&self) -> u32 {
+        0
+    }
+
+    /// Formats the unit of the given `axis` for SCMI protocol.
+    ///
+    /// Usually no need to redefine this.
+    fn format_unit(&self, axis: u32) -> u32 {
+        (self.unit_exponent(axis) as u32 & 0x1F) << 11 | u32::from(self.unit())
+    }
+
+    /// Returns SCMI description of the sensor.
+    ///
+    /// Usually no need to redefine this.
+    fn description_get(&self) -> DeviceResult {
+        // Continuous update required by Linux SCMI IIO driver
+        let low = 1 << 30;
+        let n_axes = self.number_of_axes();
+        let high = if n_axes > 0 {
+            n_axes << 16 | 1 << 8
+        } else {
+            self.format_unit(0)
+        };
+        let name = self.sensor().name.clone();
+        let values: MessageValues = vec![
+            // attributes low
+            MessageValue::Unsigned(low),
+            // attributes high
+            MessageValue::Unsigned(high),
+            // name, up to 16 bytes with final NULL (non-extended version)
+            MessageValue::String(name, MAX_SIMPLE_STRING_LENGTH),
+        ];
+        Ok(values)
+    }
+
+    /// Returns the SCMI unit of the sensor.
+    fn unit(&self) -> u8 {
+        scmi::SENSOR_UNIT_UNSPECIFIED
+    }
+
+    /// Returns the decadic exponent to apply to the sensor values.
+    fn unit_exponent(&self, _axis: u32) -> i8 {
+        0
+    }
+
+    /// Returns the prefix of axes names.
+    ///
+    /// Usually no need to redefine this.
+    fn axis_name_prefix(&self) -> String {
+        String::from("axis")
+    }
+
+    /// Returns the suffix of the given axis.
+    ///
+    /// Usually no need to redefine this.
+    fn axis_name_suffix(&self, axis: u32) -> char {
+        match axis {
+            0 => 'X',
+            1 => 'Y',
+            2 => 'Z',
+            _ => 'N', // shouldn't be reached currently
+        }
+    }
+
+    /// Returns the SCMI description of the given axis.
+    ///
+    /// Usually no need to redefine this.
+    fn axis_description(&self, axis: u32) -> Vec<MessageValue> {
+        let mut values = vec![];
+        values.push(MessageValue::Unsigned(axis)); // axis id
+        values.push(MessageValue::Unsigned(0)); // attributes low
+        values.push(MessageValue::Unsigned(self.format_unit(axis))); // attributes high
+
+        // Name in the recommended format, 16 bytes:
+        let prefix = self.axis_name_prefix();
+        let suffix = self.axis_name_suffix(axis);
+        values.push(MessageValue::String(
+            format!("{prefix}_{suffix}"),
+            MAX_SIMPLE_STRING_LENGTH,
+        ));
+        values
+    }
+
+    /// Returns the SCMI configuration of the sensor.
+    ///
+    /// The default implementation here returns just whether the sensor is
+    /// enabled or not.
+    fn config_get(&self) -> DeviceResult {
+        let config = u32::from(self.sensor().enabled);
+        Result::Ok(vec![MessageValue::Unsigned(config)])
```

Comment:
```suggestion
        Ok(vec![MessageValue::Unsigned(config)])
```
I don't think the `Result::` is needed here and elsewhere

Reply:
Done.